### PR TITLE
Potential memory leaks

### PR DESCRIPTION
**Pull request by endJunction**
Main interface change in TemplateVec is the usage of unique pointer for the name-id map.
The potential memory leakage is reported by clang's static code analyzer (clang-tidy clang-analyzer-cplusplus.NewDeleteLeaks).

I suggest commit-wise review.

### CODE REVIEW

File: GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp [hunk 1]
```diff
@@ -70,41 +71,66 @@ int XmlGmlInterface::readFile(const QString &fileName)
...
             }
             else
                 gliName = type_node.toElement().text().toStdString();
         else if (nodeName.compare("points") == 0)
         {
-            readPoints(type_node, points.get(), pnt_names);
-            _geo_objs.addPointVec(std::move(points), gliName, pnt_names);
+            readPoints(type_node, points.get(), pnt_names.get());
+
+            // if names-map is empty, set it to NULL because it is not needed
```
Comment (TomFischer) [inline GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp]: Minor: `NULL` -> `nullptr`, also in the comments below.

File: GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp [hunk 2]
```diff
@@ -70,41 +71,66 @@ int XmlGmlInterface::readFile(const QString &fileName)
...
+            if (sfc_names->empty())
+            {
+                sfc_names.reset(nullptr);
+            }
+        }
     }
 
     if (polylines->empty())
-        deletePolylines(std::move(polylines), ply_names);
+        deletePolylines(std::move(polylines));
```
Comment (TomFischer) [inline GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp]: There aren't any `Polyline`s in the vector `polylines`. Is it necessary to call `delete Polylines`?

File: GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp [hunk 3]
```diff
@@ -70,41 +71,66 @@ int XmlGmlInterface::readFile(const QString &fileName)
...
     else
-        _geo_objs.addPolylineVec(std::move(polylines), gliName, ply_names);
+        _geo_objs.addPolylineVec(std::move(polylines), gliName,
+                                 std::move(ply_names));
 
     if (surfaces->empty())
-        deleteSurfaces(std::move(surfaces), sfc_names);
+        deleteSurfaces(std::move(surfaces));
```
Comment (TomFischer) [inline GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp]: There aren't any `Surface` objects in the vector `surfaces`. Is it necessary to call `deleteSurfaces`?

File: GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp [hunk 4]
```diff
@@ -24,6 +24,34 @@
...
+
+void deletePolylines(std::unique_ptr<std::vector<GeoLib::Polyline*>> polylines)
+{
+    for (GeoLib::Polyline* line : *polylines)
+        delete line;
+}
+
+void deleteSurfaces(std::unique_ptr<std::vector<GeoLib::Surface*>> surfaces)
+{
+    for (GeoLib::Surface* line : *surfaces)
```
Comment (TomFischer) [inline GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp]: It is better to use `surface` instead of `line`.

**Comment by endJunction** [comment]
Just rebased, ready for merge or review...

**Pull request merged**